components/tooltip/tooltip.js

```diff
@@ -61,975 +31,1485 @@ const tooltipInteractiveRoles = {
+				 * @type {boolean}
+				 */
+				forceShow: { type: Boolean, attribute: 'force-show' },
+				/**
+				 * ADVANCED: Accessibility type for the tooltip to specify whether it is the primary label for the target or a secondary descriptor.
+				 * @type {'label'|'descriptor'}
+				 */
+				forType: { type: String, attribute: 'for-type' },
+				/**
+				 * Adjust the size of the gap between the tooltip and its target (px)
+				 * @type {number}
+				 */
+				offset: { type: Number },
+				/**
+				 * @ignore
+				 */
+				showing: { type: Boolean, reflect: true },
+				/**
+				 * ADVANCED: Only show the tooltip if we detect the target element is truncated
+				 * @type {boolean}
+				 */
+				showTruncatedOnly: { type: Boolean, attribute: 'show-truncated-only' }
+			};
+		}
 
-			.d2l-tooltip-pointer {
-				border: 1px solid transparent; /* fixes a webket clipping defect */
-				box-sizing: border-box;
-				display: inline-block;
-				height: ${pointerLength}px;
-				position: absolute;
-				width: ${pointerLength}px;
-				z-index: 1;
-			}
+		static get styles() {
+			return [super.styles];
+		}
 
-			:host([_open-dir="top"]) .d2l-tooltip-pointer,
-			:host([_open-dir="bottom"]) .d2l-tooltip-pointer {
-				left: calc(50% - ${pointerLength / 2}px);
-			}
+		constructor() {
+			super();
+			this.announced = false;
+			this.closeOnClick = false;
+			this.delay = 300;
+			this.disableFocusLock = false;
+			this.forceShow = false;
+			this.forType = 'descriptor';
+			this.offset = 10;
+			this.showTruncatedOnly = false;
+
+			this.#handleTargetBlurBound = this.#handleTargetBlur.bind(this);
+			this.#handleTargetClickBound = this.#handleTargetClick.bind(this);
+			this.#handleTargetFocusBound = this.#handleTargetFocus.bind(this);
+			this.#handleTargetMouseEnterBound = this.#handleTargetMouseEnter.bind(this);
+			this.#handleTargetMouseLeaveBound = this.#handleTargetMouseLeave.bind(this);
+			this.#handleTargetResizeBound = this.#handleTargetResize.bind(this);
+			this.#handleTargetTouchEndBound = this.#handleTargetTouchEnd.bind(this);
+			this.#handleTargetTouchStartBound = this.#handleTargetTouchStart.bind(this);
+
+			this.#dismissibleId = null;
+			this.#isFocusing = false;
+			this.#isHovering = false;
+			this.#isHoveringTooltip = false;
+			this.#isTruncating = false;
+			this.#mouseLeftTooltip = false;
+			this.#resizeRunSinceTruncationCheck = false;
 
-			:host([_open-dir="top"][align="start"]) .d2l-tooltip-pointer,
-			:host([_open-dir="bottom"][align="start"]) .d2l-tooltip-pointer,
-			:host([_open-dir="top"][align="end"][dir="rtl"]) .d2l-tooltip-pointer,
-			:host([_open-dir="bottom"][align="end"][dir="rtl"]) .d2l-tooltip-pointer {
-				left: ${contentHorizontalPadding + (pointerRotatedLength - pointerLength) / 2}px; /* needed for browsers that don't support min like Legacy-Edge */
-				left: min(${contentHorizontalPadding + (pointerRotatedLength - pointerLength) / 2}px, calc(50% - ${pointerLength / 2}px));
-				right: auto;
-			}
+			/*
```

Comment:
Did you mean to leave this commented out?

Reply:
Oh yeah... that'll get cleaned up/uncommented shortly as I add the actual tooltip styles. Copied from the original as a note to myself of things to deal with.